web/src/classic/core/mantle/data/gpx.ts

```diff
@@ -14,111 +13,547 @@ export async function fetchGPXfile(
+          linkElement !== null
+            ? {
+                href: linkElement.getAttribute("href") ?? "",
+                text: getElementValue(linkElement, "text"),
+                type: getElementValue(linkElement, "type"),
+              }
+            : null,
+      };
+    }
+
+    // Parse and store the link element and its associated data
+    const linkElement = queryDirectSelector(metadata, "link");
+    if (linkElement !== null) {
+      output.metadata.link = {
+        href: linkElement.getAttribute("href") ?? "",
+        text: getElementValue(linkElement, "text"),
+        type: getElementValue(linkElement, "type"),
+      };
+    }
+  }
+
+  // Parse and store all waypoints
+  const waypoints = Array.from(output.xml.querySelectorAll("wpt"));
+  for (const waypoint of waypoints) {
+    const point: Waypoint = {
+      name: getElementValue(waypoint, "name"),
+      symbol: getElementValue(waypoint, "sym"),
+      latitude: parseFloat(waypoint.getAttribute("lat") ?? ""),
+      longitude: parseFloat(waypoint.getAttribute("lon") ?? ""),
+      elevation: null,
+      comment: getElementValue(waypoint, "cmt"),
+      description: getElementValue(waypoint, "desc"),
+      time: null,
+    };
+
+    const rawElevation = parseFloat(getElementValue(waypoint, "ele"));
+    point.elevation = isNaN(rawElevation) ? null : rawElevation;
+
+    const rawTime = getElementValue(waypoint, "time");
+    point.time = rawTime == null ? null : new Date(rawTime);
+
+    output.waypoints.push(point);
+  }
+
+  const routes = Array.from(output.xml.querySelectorAll("rte"));
+  for (const routeElement of routes) {
+    const route: Route = {
+      name: getElementValue(routeElement, "name"),
+      comment: getElementValue(routeElement, "cmt"),
+      description: getElementValue(routeElement, "desc"),
+      src: getElementValue(routeElement, "src"),
+      number: getElementValue(routeElement, "number"),
+      type: null,
+      link: null,
+      points: [],
+      distance: {
+        cumulative: [],
+        total: 0,
+      },
+      elevation: {
+        maximum: null,
+        minimum: null,
+        average: null,
+        positive: null,
+        negative: null,
+      },
+      slopes: [],
+    };
+
+    const type = queryDirectSelector(routeElement, "type");
+    route.type = type !== null ? type.innerHTML : null;
+
+    // Parse and store the link and its associated data
+    const linkElement = routeElement.querySelector("link");
+    if (linkElement !== null) {
+      route.link = {
+        href: linkElement.getAttribute("href") ?? "",
+        text: getElementValue(linkElement, "text"),
+        type: getElementValue(linkElement, "type"),
+      };
+    }
+
+    // Parse and store all points in the route
+    const routePoints = Array.from(routeElement.querySelectorAll("rtept"));
+    for (const routePoint of routePoints) {
+      const point: Point = {
+        latitude: parseFloat(routePoint.getAttribute("lat") ?? ""),
```

Comment:
`parseFloat("")` returns NaN. I think we should handle this case with more safer way.

Reply:
https://github.com/reearth/reearth/pull/590/commits/686efefaeb0ea8739da9f6bb27464449c4f54428 Should handle it.